Index: linux-stage/fs/ext4/ext4.h
===================================================================
--- linux-stage.orig/fs/ext4/ext4.h	2011-04-19 01:02:34.000000000 +0800
+++ linux-stage/fs/ext4/ext4.h	2011-04-19 01:02:34.000000000 +0800
@@ -55,6 +55,14 @@
 #define ext4_debug(f, a...)	do {} while (0)
 #endif
 
+#ifndef O_32BITHASH
+# define O_32BITHASH	0x10000000
+#endif
+
+#ifndef O_64BITHASH
+# define O_64BITHASH	0x20000000
+#endif
+
 #define HAVE_DISK_INODE_VERSION
 
 /* data type for block offset of block group */
